File: Livrili-main/Frontend/src/app/shared/models/product.ts
import { Account } from "./Account"
import { Brand } from "./Brand"
import { Category } from "./Category"

export interface Product{
    id?: number,
    name?:string,
    description?:string,

    price?:number,
    quantity?:number,
    rating?:number,
    numberRatings?:number,
    image?:string,
    category?:Category,
    brand?:Brand,
    seller?:Account
  


}

File: Livrili-main/Backend/MarketplaceBackend-main/src/main/java/com/example/marketplace/model/Product.java
package com.example.marketplace.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String description;
    private double price;
    private String image;
    private int quantity=0;
    private double rating=0;
    private int numberRatings=0;
    @ManyToOne
    private Category category;
    @ManyToOne
    private Brand brand;
    @OneToMany(mappedBy = "product")
    @JsonIgnore
    private List<OrderItem> orderItems;
    @ManyToOne
    private Account seller;



}
